src/BaGetter.Database.Sqlite/SqliteContext.cs

```diff
@@ -45,5 +50,49 @@ protected override void OnModelCreating(ModelBuilder builder)
                 .Property(f => f.Moniker)
                 .HasColumnType("TEXT COLLATE NOCASE");
         }
+
+        public override async Task RunMigrationsAsync(CancellationToken cancellationToken)
+        {
+            if (Database.GetDbConnection() is SqliteConnection connection)
+            {
+                /* Create the folder of the Sqlite blob if it does not exist. */
+                CreateSqliteDataSourceDirectory(connection);
+            }
+
+            await base.RunMigrationsAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates directories specified in the Database::ConnectionString config for the Sqlite database file.
+        /// Respects the difference of root and relative paths.
+        /// </summary>
+        /// <param name="connection">Instance of the <see cref="SqliteConnection"/>.</param>
+        private static void CreateSqliteDataSourceDirectory(SqliteConnection connection)
+        {
+            var configDataSource = connection.DataSource;
+            var pathToCreate = Path.IsPathRooted(configDataSource)
+                // root path
+                ? Path.GetDirectoryName(configDataSource)
+                // relative path
+                : CreateFullDataSourcePath();
```

Comment:
Why is this split even necessary? As far as I can see in a quick test, `Directory.CreateDirectory` will create both absolute and relative paths you throw at it, starting at the present working directory.

Reply:
Thanks for looking into this. I've tested again with your suggested changes and it worked flawlessly. I don't know what setup i had (under Windows) where it did not work so i added the switching.
The CreateDirectory() implementation keeps track of relative or root paths itself and worked without problems under Linux and Windows.